### PR TITLE
AFGS-525 Sending a letter when the claimant's phone or email mismatches.

### PR DESCRIPTION
**Pull request by dwybourn**
One of two different letters is sent depending on whether or not the benefit agency has all of the children's dob declared by the claimant.

### CODE REVIEW

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java [hunk 1]
```diff
@@ -123,6 +123,7 @@ void shouldSendWeWillLetYouKnowEmailForEligibleApplicantWhoseEmailOrPhoneDontMat
         assertThat(payment.getPaymentAmountInPence()).isEqualTo(expectedEntitlement.getTotalVoucherValueInPence());
 
         assertThatEmailWithNameOnlyWasSent(claim, PENDING_DECISION);
+        assertThatLetterWithAddressOnlyWasSent(claim, LetterType.INSTANT_SUCCESS_CHILDREN_MATCH);
```
Comment (YetAnotherMatt) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: We'll need another test for the alternative letter

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimMessageSender.java [hunk 2]
```diff
@@ -88,4 +88,14 @@ public void sendUpdateYourAddressLetterMessage(Claim claim) {
         MessagePayload payload = buildLetterPayloadWithAddressOnly(claim, LetterType.UPDATE_YOUR_ADDRESS);
         messageQueueClient.sendMessage(payload, SEND_LETTER);
     }
+
+    public void sendInstantSuccessChildrenMatchLetterMessage(Claim claim) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimMessageSender.java]: We've got 3 letter methods here now that only vary by `LetterType`, can we refactor into a single method please?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java [hunk 3]
```diff
@@ -80,12 +82,25 @@ private void sendMessagesForNewClaim(EligibilityAndEntitlementDecision decision,
...
         claimMessageSender.sendReportClaimMessage(claim, identityAndEligibilityResponse, ClaimAction.NEW);
     }
 
+    private void sendMessagesForPhoneOrEmailMismatch(CombinedIdentityAndEligibilityResponse identityAndEligibilityResponse, Claim claim) {
+        claimMessageSender.sendDecisionPendingEmailMessage(claim);
+
+        List<LocalDate> childrenDobFromBenefitAgency = identityAndEligibilityResponse.getDobOfChildrenUnder4();
+        List<LocalDate> declaredChildrenDob = claim.getClaimant().getInitiallyDeclaredChildrenDob();
+
+        if (childrenDobFromBenefitAgency.containsAll(declaredChildrenDob)) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: This doesn't cater for there being more children returned from DWP than they have declared.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java (hunk 1 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: This is covered in the parameterized unit test. I'd argue that we don't need both options at integration level.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java (hunk 3 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: We said in that scenario we'll still return children match, there is a unit test covering that scenario. @YetAnotherMatt can you confirm that's correct?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimServiceTest.java [hunk 4]
```diff
@@ -122,31 +126,55 @@ void shouldSaveNewEligibleClaimantAndSendMessages() {
         verifyNoMoreInteractions(claimMessageSender);
     }
 
-    @Test
-    void shouldSaveNewEligibleClaimantAndSendWeWillLetYouKnowEmailWhenEmailOrPhoneMismatch() {
+    @ParameterizedTest
+    @MethodSource("emailOrPhoneMismatchArguments")
+    void shouldSaveNewEligibleClaimantAndSendWeWillLetYouKnowEmailAndInstantSuccessLetterWhenEmailOrPhoneMismatch(VerificationOutcome emailVerification,
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimServiceTest.java]: Wow, I approve!!! :-)

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java (hunk 3 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: That is correct, yes. DWP reporting extra children is ok.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimMessageSender.java (hunk 2 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimMessageSender.java]: Refactored into one method.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: Given this affects the external integration (sending a letter), I'd argue it is worthwhile

Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTestsWithScheduledServices.java]: Made the test parameterized.

**Pull request merged**